parlai/mturk/core/agents.py

```diff
@@ -31,6 +31,7 @@
 polling_interval = 1 # in seconds
 create_hit_type_lock = threading.Lock()
 local_db_lock = threading.Lock()
+task_dir = os.getcwd()
```

Comment:
I agreed - os.getcwd() is not guaranteed to return the directory that the file lives in. Changing it to be using the method you suggested.